modules/systemd-journal/journal-reader.c

```diff
@@ -585,47 +585,61 @@ _add_poll_events(JournalReader *self)
   return TRUE;
 }
 
-static gint
-_journal_open(JournalReader *self)
-{
 #if SYSLOG_NG_HAVE_JOURNAL_NAMESPACES
-  gint journal_flags = SD_JOURNAL_LOCAL_ONLY;
-  gchar *journal_namespace = NULL;
-
-  if (strcmp(self->options->namespace, "*") == 0)
+static void
+_journal_process_namespace(gchar *namespace_option, gchar **journal_namespace, gint *journal_flags)
+{
+  if (strcmp(namespace_option, "*") == 0)
     {
-      journal_flags |= SD_JOURNAL_ALL_NAMESPACES;
-      if (strlen(self->options->namespace) > 1)
+      *journal_flags |= SD_JOURNAL_ALL_NAMESPACES;
+      if (strlen(namespace_option) > 1)
         {
           msg_warning("namespace('*'): discarding everything after the asterisk");
         }
     }
-  else if (strncmp(self->options->namespace, "+", 1) == 0)
+  else if (strncmp(namespace_option, "+", 1) == 0)
     {
-      journal_flags |= SD_JOURNAL_INCLUDE_DEFAULT_NAMESPACE;
-      if (strlen(self->options->namespace) > 1)
+      *journal_flags |= SD_JOURNAL_INCLUDE_DEFAULT_NAMESPACE;
+      if (strlen(namespace_option) > 1)
         {
-          journal_namespace = self->options->namespace + 1;
+          *journal_namespace = namespace_option + 1;
         }
     }
   else
     {
-      if (strlen(self->options->namespace) > 0)
+      if (strlen(namespace_option) > 0)
         {
-          journal_namespace = self->options->namespace;
+          *journal_namespace = namespace_option;
         }
     }
+}
+#endif
+
+static gint
+_journal_open(JournalReader *self)
+{
+  gint journal_flags = SD_JOURNAL_LOCAL_ONLY;
 
-  if (journal_namespace)
+#if SYSLOG_NG_HAVE_JOURNAL_NAMESPACES
+  gchar *journal_namespace = NULL;
+  _journal_process_namespace(self->options->namespace, &journal_namespace, &journal_flags);
+  return journald_open_namespace(self->journal, journal_namespace, journal_flags);
+#else
+  return journald_open(self->journal, journal_flags);
+#endif
+}
+
+static void
+_init_persist_name(JournalReader *self)
+{
+  if (SYSLOG_NG_HAVE_JOURNAL_NAMESPACES && (strcmp(self->options->namespace, "*") != 0))
```

Comment:
Alternatives:
1. let the `*` be the default value of `self->options->namespace` (and set it to that even if there is no namespace support), as a result here `SYSLOG_NG_HAVE_JOURNAL_NAMESPACES` check is not needed.
2. the `SYSLOG_NG_HAVE_JOURNAL_NAMESPACES` should be used in preprocessor only not in condition

Reply:
Thank You!